app/views/spree/shared/_nav_bar.html.erb

```diff
@@ -104,13 +103,16 @@
               <% end %>
             <% end %>
           </div>
-          <%= link_to cart_path, :class =>'nav-cart', id: 'link-to-cart'  do%>
-            <span id="cart">cart </span>
-            <small class="cart-counter">4</small>     
-          <% end %>
+          <a id="shopping_cart_bag" class="nav-cart" href="#" >
+            <span>cart </span>
+            <small class="cart-counter"><%= current_order.line_items.count %></small>
```

Comment:
Please add current_order presence check like this.

current_order.present? ? current_order.line_items.count : 0